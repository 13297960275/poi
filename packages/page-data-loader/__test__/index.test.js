const path = require('path')
const { promisify } = require('util')
const webpack = require('webpack')
const { VueLoaderPlugin } = require('vue-loader')
const req = require('require-from-string')

it('works', async () => {
  const compiler = webpack({
    mode: 'development',
    devtool: false,
    entry: path.join(__dirname, 'fixtures/index.js'),
    output: {
      path: '/',
      filename: 'main.js',
      libraryTarget: 'commonjs2'
    },
    module: {
      rules: [
        {
          resourceQuery: /blockType=page-data/,
          loader: require.resolve('..'),
          options: {
            cacheDir: path.join(__dirname, '.cache')
          }
        },
        {
          test: /\.vue$/,
          loader: 'vue-loader'
        }
      ]
    },
    plugins: [new VueLoaderPlugin()]
  })
  const mfs = new webpack.MemoryOutputFileSystem()
  compiler.outputFileSystem = mfs
  const stats = await promisify(compiler.run.bind(compiler))()
  if (stats.hasErrors() || stats.hasWarnings()) {
    throw new Error(stats.toString())
  }
  const code = mfs.readFileSync('/main.js', 'utf8')
  expect(code).toMatchSnapshot()
  const exported = req(code)
  expect(exported.foo.__pageData()).toMatchSnapshot('foo')
  expect(exported.user.__pageData({ user: 'egoist' })).toMatchSnapshot(
    'user egoist'
  )
  expect(exported.user.__pageData({ user: 'cristiano' })).toMatchSnapshot(
    'user cristiano'
  )
})
