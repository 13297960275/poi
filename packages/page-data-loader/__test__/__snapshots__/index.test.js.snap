// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"module.exports =
/******/ (function(modules) { // webpackBootstrap
/******/ 	// The module cache
/******/ 	var installedModules = {};
/******/
/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {
/******/
/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId]) {
/******/ 			return installedModules[moduleId].exports;
/******/ 		}
/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};
/******/
/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
/******/
/******/ 		// Flag the module as loaded
/******/ 		module.l = true;
/******/
/******/ 		// Return the exports of the module
/******/ 		return module.exports;
/******/ 	}
/******/
/******/
/******/ 	// expose the modules object (__webpack_modules__)
/******/ 	__webpack_require__.m = modules;
/******/
/******/ 	// expose the module cache
/******/ 	__webpack_require__.c = installedModules;
/******/
/******/ 	// define getter function for harmony exports
/******/ 	__webpack_require__.d = function(exports, name, getter) {
/******/ 		if(!__webpack_require__.o(exports, name)) {
/******/ 			Object.defineProperty(exports, name, { enumerable: true, get: getter });
/******/ 		}
/******/ 	};
/******/
/******/ 	// define __esModule on exports
/******/ 	__webpack_require__.r = function(exports) {
/******/ 		if(typeof Symbol !== 'undefined' && Symbol.toStringTag) {
/******/ 			Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });
/******/ 		}
/******/ 		Object.defineProperty(exports, '__esModule', { value: true });
/******/ 	};
/******/
/******/ 	// create a fake namespace object
/******/ 	// mode & 1: value is a module id, require it
/******/ 	// mode & 2: merge all properties of value into the ns
/******/ 	// mode & 4: return value when already ns object
/******/ 	// mode & 8|1: behave like require
/******/ 	__webpack_require__.t = function(value, mode) {
/******/ 		if(mode & 1) value = __webpack_require__(value);
/******/ 		if(mode & 8) return value;
/******/ 		if((mode & 4) && typeof value === 'object' && value && value.__esModule) return value;
/******/ 		var ns = Object.create(null);
/******/ 		__webpack_require__.r(ns);
/******/ 		Object.defineProperty(ns, 'default', { enumerable: true, value: value });
/******/ 		if(mode & 2 && typeof value != 'string') for(var key in value) __webpack_require__.d(ns, key, function(key) { return value[key]; }.bind(null, key));
/******/ 		return ns;
/******/ 	};
/******/
/******/ 	// getDefaultExport function for compatibility with non-harmony modules
/******/ 	__webpack_require__.n = function(module) {
/******/ 		var getter = module && module.__esModule ?
/******/ 			function getDefault() { return module['default']; } :
/******/ 			function getModuleExports() { return module; };
/******/ 		__webpack_require__.d(getter, 'a', getter);
/******/ 		return getter;
/******/ 	};
/******/
/******/ 	// Object.prototype.hasOwnProperty.call
/******/ 	__webpack_require__.o = function(object, property) { return Object.prototype.hasOwnProperty.call(object, property); };
/******/
/******/ 	// __webpack_public_path__
/******/ 	__webpack_require__.p = \\"\\";
/******/
/******/
/******/ 	// Load entry module and return exports
/******/ 	return __webpack_require__(__webpack_require__.s = \\"./packages/page-data-loader/__test__/fixtures/index.js\\");
/******/ })
/************************************************************************/
/******/ ({

/***/ \\"./node_modules/vue-loader/lib/runtime/componentNormalizer.js\\":
/*!********************************************************************!*\\\\
  !*** ./node_modules/vue-loader/lib/runtime/componentNormalizer.js ***!
  \\\\********************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"default\\", function() { return normalizeComponent; });
/* globals __VUE_SSR_CONTEXT__ */

// IMPORTANT: Do NOT use ES2015 features in this file (except for modules).
// This module is a runtime utility for cleaner component module output and will
// be included in the final webpack user bundle.

function normalizeComponent (
  scriptExports,
  render,
  staticRenderFns,
  functionalTemplate,
  injectStyles,
  scopeId,
  moduleIdentifier, /* server only */
  shadowMode /* vue-cli only */
) {
  // Vue.extend constructor export interop
  var options = typeof scriptExports === 'function'
    ? scriptExports.options
    : scriptExports

  // render functions
  if (render) {
    options.render = render
    options.staticRenderFns = staticRenderFns
    options._compiled = true
  }

  // functional template
  if (functionalTemplate) {
    options.functional = true
  }

  // scopedId
  if (scopeId) {
    options._scopeId = 'data-v-' + scopeId
  }

  var hook
  if (moduleIdentifier) { // server build
    hook = function (context) {
      // 2.3 injection
      context =
        context || // cached call
        (this.$vnode && this.$vnode.ssrContext) || // stateful
        (this.parent && this.parent.$vnode && this.parent.$vnode.ssrContext) // functional
      // 2.2 with runInNewContext: true
      if (!context && typeof __VUE_SSR_CONTEXT__ !== 'undefined') {
        context = __VUE_SSR_CONTEXT__
      }
      // inject component styles
      if (injectStyles) {
        injectStyles.call(this, context)
      }
      // register component module identifier for async chunk inferrence
      if (context && context._registeredComponents) {
        context._registeredComponents.add(moduleIdentifier)
      }
    }
    // used by ssr in case component is cached and beforeCreate
    // never gets called
    options._ssrRegister = hook
  } else if (injectStyles) {
    hook = shadowMode
      ? function () { injectStyles.call(this, this.$root.$options.shadowRoot) }
      : injectStyles
  }

  if (hook) {
    if (options.functional) {
      // for template-only hot-reload because in that case the render fn doesn't
      // go through the normalizer
      options._injectStyles = hook
      // register for functioal component in vue file
      var originalRender = options.render
      options.render = function renderWithStyleInjection (h, context) {
        hook.call(context)
        return originalRender(h, context)
      }
    } else {
      // inject component registration as beforeCreate hook
      var existing = options.beforeCreate
      options.beforeCreate = existing
        ? [].concat(existing, hook)
        : [hook]
    }
  }

  return {
    exports: scriptExports,
    options: options
  }
}


/***/ }),

/***/ \\"./packages/page-data-loader/__test__/.cache sync recursive ^\\\\\\\\.\\\\\\\\/packages\\\\\\\\-page\\\\\\\\-data\\\\\\\\-loader\\\\\\\\-__test__\\\\\\\\-fixtures\\\\\\\\-\\\\\\\\[user\\\\\\\\]\\\\\\\\-vue__.*\\\\\\\\.json$\\":
/*!****************************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/.cache sync ^\\\\.\\\\/packages\\\\-page\\\\-data\\\\-loader\\\\-__test__\\\\-fixtures\\\\-\\\\[user\\\\]\\\\-vue__.*\\\\.json$ ***!
  \\\\****************************************************************************************************************************************/
/*! no static exports found */
/***/ (function(module, exports, __webpack_require__) {

var map = {
	\\"./packages-page-data-loader-__test__-fixtures-[user]-vue__[default].json\\": \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__[default].json\\",
	\\"./packages-page-data-loader-__test__-fixtures-[user]-vue__cristiano.json\\": \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__cristiano.json\\",
	\\"./packages-page-data-loader-__test__-fixtures-[user]-vue__egoist.json\\": \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__egoist.json\\"
};


function webpackContext(req) {
	var id = webpackContextResolve(req);
	return __webpack_require__(id);
}
function webpackContextResolve(req) {
	var id = map[req];
	if(!(id + 1)) { // check for number or string
		var e = new Error(\\"Cannot find module '\\" + req + \\"'\\");
		e.code = 'MODULE_NOT_FOUND';
		throw e;
	}
	return id;
}
webpackContext.keys = function webpackContextKeys() {
	return Object.keys(map);
};
webpackContext.resolve = webpackContextResolve;
module.exports = webpackContext;
webpackContext.id = \\"./packages/page-data-loader/__test__/.cache sync recursive ^\\\\\\\\.\\\\\\\\/packages\\\\\\\\-page\\\\\\\\-data\\\\\\\\-loader\\\\\\\\-__test__\\\\\\\\-fixtures\\\\\\\\-\\\\\\\\[user\\\\\\\\]\\\\\\\\-vue__.*\\\\\\\\.json$\\";

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__[default].json\\":
/*!**************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__[default].json ***!
  \\\\**************************************************************************************************************************/
/*! exports provided: text, default */
/***/ (function(module) {

module.exports = {\\"text\\":\\"hello world\\"};

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__cristiano.json\\":
/*!**************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__cristiano.json ***!
  \\\\**************************************************************************************************************************/
/*! exports provided: text, name, age, default */
/***/ (function(module) {

module.exports = {\\"text\\":\\"hello world\\",\\"name\\":\\"Cristiano Ronaldo\\",\\"age\\":33};

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__egoist.json\\":
/*!***********************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-[user]-vue__egoist.json ***!
  \\\\***********************************************************************************************************************/
/*! exports provided: text, name, age, default */
/***/ (function(module) {

module.exports = {\\"text\\":\\"hello world\\",\\"name\\":\\"EGOIST\\",\\"age\\":24};

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-foo-vue__[default].json\\":
/*!***********************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-foo-vue__[default].json ***!
  \\\\***********************************************************************************************************************/
/*! exports provided: text, default */
/***/ (function(module) {

module.exports = {\\"text\\":\\"world\\"};

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/[user].vue\\":
/*!****************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/[user].vue ***!
  \\\\****************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _user_vue_vue_type_template_id_40542ea2___WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./[user].vue?vue&type=template&id=40542ea2& */ \\"./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=template&id=40542ea2&\\");
/* harmony import */ var _node_modules_vue_loader_lib_runtime_componentNormalizer_js__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ../../../../node_modules/vue-loader/lib/runtime/componentNormalizer.js */ \\"./node_modules/vue-loader/lib/runtime/componentNormalizer.js\\");
/* harmony import */ var _user_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ./[user].vue?vue&type=custom&index=0&blockType=page-data */ \\"./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=custom&index=0&blockType=page-data\\");

var script = {}


/* normalize component */

var component = Object(_node_modules_vue_loader_lib_runtime_componentNormalizer_js__WEBPACK_IMPORTED_MODULE_1__[\\"default\\"])(
  script,
  _user_vue_vue_type_template_id_40542ea2___WEBPACK_IMPORTED_MODULE_0__[\\"render\\"],
  _user_vue_vue_type_template_id_40542ea2___WEBPACK_IMPORTED_MODULE_0__[\\"staticRenderFns\\"],
  false,
  null,
  null,
  null
  
)

/* custom blocks */

if (typeof _user_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_2__[\\"default\\"] === 'function') Object(_user_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_2__[\\"default\\"])(component)

/* hot reload */
if (false) { var api; }
component.options.__file = \\"packages/page-data-loader/__test__/fixtures/[user].vue\\"
/* harmony default export */ __webpack_exports__[\\"default\\"] = (component.exports);

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=custom&index=0&blockType=page-data\\":
/*!************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=custom&index=0&blockType=page-data ***!
  \\\\************************************************************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _lib_index_js_ref_0_node_modules_vue_loader_lib_index_js_vue_loader_options_user_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! -!../../lib??ref--0!../../../../node_modules/vue-loader/lib??vue-loader-options!./[user].vue?vue&type=custom&index=0&blockType=page-data */ \\"./packages/page-data-loader/lib/index.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=custom&index=0&blockType=page-data\\");
/* empty/unused harmony star reexport */ /* harmony default export */ __webpack_exports__[\\"default\\"] = (_lib_index_js_ref_0_node_modules_vue_loader_lib_index_js_vue_loader_options_user_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_0__[\\"default\\"]); 

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=template&id=40542ea2&\\":
/*!***********************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=template&id=40542ea2& ***!
  \\\\***********************************************************************************************/
/*! exports provided: render, staticRenderFns */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _node_modules_vue_loader_lib_loaders_templateLoader_js_vue_loader_options_node_modules_vue_loader_lib_index_js_vue_loader_options_user_vue_vue_type_template_id_40542ea2___WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! -!../../node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!../../../../node_modules/vue-loader/lib??vue-loader-options!./[user].vue?vue&type=template&id=40542ea2& */ \\"./packages/page-data-loader/node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=template&id=40542ea2&\\");
/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\"render\\", function() { return _node_modules_vue_loader_lib_loaders_templateLoader_js_vue_loader_options_node_modules_vue_loader_lib_index_js_vue_loader_options_user_vue_vue_type_template_id_40542ea2___WEBPACK_IMPORTED_MODULE_0__[\\"render\\"]; });

/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\"staticRenderFns\\", function() { return _node_modules_vue_loader_lib_loaders_templateLoader_js_vue_loader_options_node_modules_vue_loader_lib_index_js_vue_loader_options_user_vue_vue_type_template_id_40542ea2___WEBPACK_IMPORTED_MODULE_0__[\\"staticRenderFns\\"]; });



/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/foo.vue\\":
/*!*************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/foo.vue ***!
  \\\\*************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _foo_vue_vue_type_template_id_725187aa___WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./foo.vue?vue&type=template&id=725187aa& */ \\"./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=template&id=725187aa&\\");
/* harmony import */ var _node_modules_vue_loader_lib_runtime_componentNormalizer_js__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ../../../../node_modules/vue-loader/lib/runtime/componentNormalizer.js */ \\"./node_modules/vue-loader/lib/runtime/componentNormalizer.js\\");
/* harmony import */ var _foo_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ./foo.vue?vue&type=custom&index=0&blockType=page-data */ \\"./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=custom&index=0&blockType=page-data\\");

var script = {}


/* normalize component */

var component = Object(_node_modules_vue_loader_lib_runtime_componentNormalizer_js__WEBPACK_IMPORTED_MODULE_1__[\\"default\\"])(
  script,
  _foo_vue_vue_type_template_id_725187aa___WEBPACK_IMPORTED_MODULE_0__[\\"render\\"],
  _foo_vue_vue_type_template_id_725187aa___WEBPACK_IMPORTED_MODULE_0__[\\"staticRenderFns\\"],
  false,
  null,
  null,
  null
  
)

/* custom blocks */

if (typeof _foo_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_2__[\\"default\\"] === 'function') Object(_foo_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_2__[\\"default\\"])(component)

/* hot reload */
if (false) { var api; }
component.options.__file = \\"packages/page-data-loader/__test__/fixtures/foo.vue\\"
/* harmony default export */ __webpack_exports__[\\"default\\"] = (component.exports);

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=custom&index=0&blockType=page-data\\":
/*!*********************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=custom&index=0&blockType=page-data ***!
  \\\\*********************************************************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _lib_index_js_ref_0_node_modules_vue_loader_lib_index_js_vue_loader_options_foo_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! -!../../lib??ref--0!../../../../node_modules/vue-loader/lib??vue-loader-options!./foo.vue?vue&type=custom&index=0&blockType=page-data */ \\"./packages/page-data-loader/lib/index.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=custom&index=0&blockType=page-data\\");
/* empty/unused harmony star reexport */ /* harmony default export */ __webpack_exports__[\\"default\\"] = (_lib_index_js_ref_0_node_modules_vue_loader_lib_index_js_vue_loader_options_foo_vue_vue_type_custom_index_0_blockType_page_data__WEBPACK_IMPORTED_MODULE_0__[\\"default\\"]); 

/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=template&id=725187aa&\\":
/*!********************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=template&id=725187aa& ***!
  \\\\********************************************************************************************/
/*! exports provided: render, staticRenderFns */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _node_modules_vue_loader_lib_loaders_templateLoader_js_vue_loader_options_node_modules_vue_loader_lib_index_js_vue_loader_options_foo_vue_vue_type_template_id_725187aa___WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! -!../../node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!../../../../node_modules/vue-loader/lib??vue-loader-options!./foo.vue?vue&type=template&id=725187aa& */ \\"./packages/page-data-loader/node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=template&id=725187aa&\\");
/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\"render\\", function() { return _node_modules_vue_loader_lib_loaders_templateLoader_js_vue_loader_options_node_modules_vue_loader_lib_index_js_vue_loader_options_foo_vue_vue_type_template_id_725187aa___WEBPACK_IMPORTED_MODULE_0__[\\"render\\"]; });

/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\"staticRenderFns\\", function() { return _node_modules_vue_loader_lib_loaders_templateLoader_js_vue_loader_options_node_modules_vue_loader_lib_index_js_vue_loader_options_foo_vue_vue_type_template_id_725187aa___WEBPACK_IMPORTED_MODULE_0__[\\"staticRenderFns\\"]; });



/***/ }),

/***/ \\"./packages/page-data-loader/__test__/fixtures/index.js\\":
/*!**************************************************************!*\\\\
  !*** ./packages/page-data-loader/__test__/fixtures/index.js ***!
  \\\\**************************************************************/
/*! exports provided: foo, user */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony import */ var _foo_vue__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./foo.vue */ \\"./packages/page-data-loader/__test__/fixtures/foo.vue\\");
/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\"foo\\", function() { return _foo_vue__WEBPACK_IMPORTED_MODULE_0__[\\"default\\"]; });

/* harmony import */ var _user_vue__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ./[user].vue */ \\"./packages/page-data-loader/__test__/fixtures/[user].vue\\");
/* harmony reexport (safe) */ __webpack_require__.d(__webpack_exports__, \\"user\\", function() { return _user_vue__WEBPACK_IMPORTED_MODULE_1__[\\"default\\"]; });







/***/ }),

/***/ \\"./packages/page-data-loader/lib/index.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=custom&index=0&blockType=page-data\\":
/*!******************************************************************************************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/lib??ref--0!./node_modules/vue-loader/lib??vue-loader-options!./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=custom&index=0&blockType=page-data ***!
  \\\\******************************************************************************************************************************************************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);

  /* harmony default export */ __webpack_exports__[\\"default\\"] = (function (Component) {
    Component.options.__pageData = function (params) {
      return __webpack_require__(\\"./packages/page-data-loader/__test__/.cache sync recursive ^\\\\\\\\.\\\\\\\\/packages\\\\\\\\-page\\\\\\\\-data\\\\\\\\-loader\\\\\\\\-__test__\\\\\\\\-fixtures\\\\\\\\-\\\\\\\\[user\\\\\\\\]\\\\\\\\-vue__.*\\\\\\\\.json$\\")(\\"./packages-page-data-loader-__test__-fixtures-[user]-vue__\\"+params['user']+'.json')
    }
    var oldData = Component.options.data || (function () {})
    Component.options.data = function () {
      return Object.assign({}, this.$options.__pageData(this.$route.params), oldData.call(this))
    }
  });
  

/***/ }),

/***/ \\"./packages/page-data-loader/lib/index.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=custom&index=0&blockType=page-data\\":
/*!***************************************************************************************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/lib??ref--0!./node_modules/vue-loader/lib??vue-loader-options!./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=custom&index=0&blockType=page-data ***!
  \\\\***************************************************************************************************************************************************************************************************/
/*! exports provided: default */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);

  /* harmony default export */ __webpack_exports__[\\"default\\"] = (function (Component) {
    Component.options.__pageData = function (params) {
      return __webpack_require__(/*! ./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-foo-vue__[default].json */ \\"./packages/page-data-loader/__test__/.cache/packages-page-data-loader-__test__-fixtures-foo-vue__[default].json\\")
    }
    var oldData = Component.options.data || (function () {})
    Component.options.data = function () {
      return Object.assign({}, this.$options.__pageData(this.$route.params), oldData.call(this))
    }
  });
  

/***/ }),

/***/ \\"./packages/page-data-loader/node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=template&id=40542ea2&\\":
/*!*******************************************************************************************************************************************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./packages/page-data-loader/__test__/fixtures/[user].vue?vue&type=template&id=40542ea2& ***!
  \\\\*******************************************************************************************************************************************************************************************************************************************************/
/*! exports provided: render, staticRenderFns */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"render\\", function() { return render; });
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"staticRenderFns\\", function() { return staticRenderFns; });
var render = function() {
  var _vm = this
  var _h = _vm.$createElement
  var _c = _vm._self._c || _h
  return _c(\\"h1\\", [
    _vm._v(
      _vm._s(_vm.text) +
        \\", \\" +
        _vm._s(_vm.name) +
        \\" is \\" +
        _vm._s(_vm.age) +
        \\" yo\\"
    )
  ])
}
var staticRenderFns = []
render._withStripped = true



/***/ }),

/***/ \\"./packages/page-data-loader/node_modules/vue-loader/lib/loaders/templateLoader.js?!./node_modules/vue-loader/lib/index.js?!./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=template&id=725187aa&\\":
/*!****************************************************************************************************************************************************************************************************************************************************!*\\\\
  !*** ./packages/page-data-loader/node_modules/vue-loader/lib/loaders/templateLoader.js??vue-loader-options!./node_modules/vue-loader/lib??vue-loader-options!./packages/page-data-loader/__test__/fixtures/foo.vue?vue&type=template&id=725187aa& ***!
  \\\\****************************************************************************************************************************************************************************************************************************************************/
/*! exports provided: render, staticRenderFns */
/***/ (function(module, __webpack_exports__, __webpack_require__) {

\\"use strict\\";
__webpack_require__.r(__webpack_exports__);
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"render\\", function() { return render; });
/* harmony export (binding) */ __webpack_require__.d(__webpack_exports__, \\"staticRenderFns\\", function() { return staticRenderFns; });
var render = function() {
  var _vm = this
  var _h = _vm.$createElement
  var _c = _vm._self._c || _h
  return _c(\\"h1\\", [_vm._v(\\"hello \\" + _vm._s(_vm.text))])
}
var staticRenderFns = []
render._withStripped = true



/***/ })

/******/ });"
`;

exports[`works: foo 1`] = `
Object {
  "text": "world",
}
`;

exports[`works: user cristiano 1`] = `
Object {
  "age": 33,
  "name": "Cristiano Ronaldo",
  "text": "hello world",
}
`;

exports[`works: user egoist 1`] = `
Object {
  "age": 24,
  "name": "EGOIST",
  "text": "hello world",
}
`;
